pkg/operator/resourcesynccontroller/resourcesynccontroller.go

```diff
@@ -28,26 +30,26 @@ func NewResourceSyncController(
 		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.TargetNamespace, Name: "etcd-serving-ca"},
 		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.EtcdNamespaceName, Name: "etcd-serving-ca"},
 	); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	if err := resourceSyncController.SyncSecret(
 		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.TargetNamespace, Name: "etcd-client"},
 		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.EtcdNamespaceName, Name: "etcd-client"},
 	); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	if err := resourceSyncController.SyncConfigMap(
 		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.TargetNamespace, Name: "client-ca"},
 		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.GlobalMachineSpecifiedConfigNamespace, Name: "kube-apiserver-client-ca"},
 	); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	if err := resourceSyncController.SyncConfigMap(
 		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.TargetNamespace, Name: "aggregator-client-ca"},
 		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.GlobalMachineSpecifiedConfigNamespace, Name: "kube-apiserver-aggregator-client-ca"},
 	); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return resourceSyncController, nil
+	return resourceSyncController, resourcesynccontroller.NewDebugHandler(resourceSyncController), nil
```

Comment:
why do we need a return value for the handler if it is only built from the already returned `resourceSyncController` and `NewDebugHandler`?